youtube_dl/extractor/plutotv.py

```diff
@@ -0,0 +1,136 @@
+                'duration': 9000,
+            }
+        },
+    ]
+
+    def _get_video_info(self, video_json, slug, series_name=None):
+        id = video_json.get('_id', slug)
+        formats = []
+        for video_url in video_json.get('stitched', {}).get('urls', []):
+            if video_url.get('type') != 'hls':
+                continue
+            url = video_url.get('url')
+            formats.extend(
+                self._extract_m3u8_formats(
+                    url, id, 'mp4', 'm3u8_native',
+                    m3u8_id='hls', fatal=False))
+        info = {
+            'id': id,
+            'formats': formats,
+            'title': video_json.get('name'),
+            'description': video_json.get('description'),
+            'duration': float_or_none(video_json.get('duration'), scale=1000),
+        }
+        if series_name:
+            info.update({
+                'series': series_name,
+                'episode': video_json.get('name'),
+                'season_number': int_or_none(video_json.get('season')),
+                'episode_number': int_or_none(video_json.get('number')),
+            })
+        return info
+
+    def _real_extract(self, url):
+        video_type, slug = re.match(self._VALID_URL, url).groups()
+        slug_components = slug.split('/')
+        info_slug = slug_components[0]
+        video_json = self._download_json(self._INFO_URL + info_slug, info_slug,
+                                         query=self._INFO_QUERY_PARAMS)
+
+        if video_type == 'movies':
+            return self._get_video_info(video_json, slug)
+        elif video_type == 'series':
+            series_name = video_json.get('name', info_slug)
+            season_number = int_or_none(try_get(slug_components, lambda x: x[2]))
+            episode_slug = try_get(slug_components, lambda x: x[4])
+
+            videos = []
+            for season in video_json.get('seasons', []):
+                if season_number is not None and season_number != int_or_none(season.get('number')):
+                    continue
+                for episode in season.get('episodes', []):
+                    if episode_slug is not None and episode_slug != episode.get('slug'):
+                        continue
+                    videos.append(self._get_video_info(episode, episode_slug, series_name))
+            if len(videos) == 0:
+                raise ExtractorError('Failed to find any videos to extract')
+            if episode_slug is not None and len(videos) == 1:
+                return videos[0]
+            playlist_title = series_name
+            if season_number is not None:
+                playlist_title += ' - Season %d' % season_number
+            return self.playlist_result(videos,
+                                        playlist_id=video_json.get('_id', info_slug),
+                                        playlist_title=playlist_title)
+        else:
+            raise ExtractorError('Unsupported url')
```

Comment:
Does youtube-dl passes to this location? You already set here the supported url:

https://github.com/ytdl-org/youtube-dl/blob/b74e1295cfbd5c0a2d825053033af65285804246/youtube_dl/extractor/plutotv.py#L17

Reply:
Currently it should be an impossible state, but I figured I would add the extra guard here should that change in the future. At least you'll get an explicit error rather than some other obscure error (since the data won't be what we expect) or malformed data.

Reply:
Hmm, okay. Not the best soulution but it's also implemented in other extractores

Reply:
Is there a pattern you prefer? I'm open to discuss alternatives.

Reply:
This must be assert not exception.

Reply:
reworked it a bit. Done!